mesonbuild/modules/python.py

```diff
@@ -280,7 +280,18 @@ def set_env(name, value):
 import json
 import sys
 
-install_paths = sysconfig.get_paths(vars={'base': '', 'platbase': '', 'installed_base': ''})
+# On Debian derivatives, the Python interpreter shipped by the distribution uses
+# a custom install scheme, deb_system, for the system install, and changes the
+# default scheme to a custom one pointing to /usr/local and replacing
+# site-packages with dist-packages.
+# See https://github.com/mesonbuild/meson/issues/8739.
+empty_vars = {'base': '', 'platbase': '', 'installed_base': ''}
+if 'deb_system' in sysconfig.get_scheme_names():
```

Comment:
On my Ubuntu 20.04:
```
>>> sysconfig.get_scheme_names()
('nt', 'nt_user', 'osx_framework_user', 'posix_home', 'posix_prefix', 'posix_user')
```

When has deb_system been added? Seems we still need the hack at least for older versions.

Reply:
It has existed for a long time, but on older versions we may need to check `distutils.commands.install.INSTALL_SCHEMES` instead.

Reply:
Which Python version is that btw?

Reply:
I just checked, and Debian started patching `syconfig` in 3.6, which is the minimum supported version by Meson. Maybe they did not patch it in all minor versions, and you happen to be running one of those?

Reply:
I tried Ubuntu 21.10 and Debian 11, both does not have that scheme. Where did you find that trick? Is that super new in Debian experimental?

Reply:
Maybe it comes from non std modules then. I tried by spawning a docker and installed python3 package.

Reply:
I did the same thing and confirmed sysconfig is not patched.

As I understood, they were doing it since 3.6, maybe I am mistaken.
https://salsa.debian.org/cpython-team/python3/-/blob/python3.6/debian/patches/sysconfig-debian-schemes.diff

Anyway, `distutils.command.install.INSTALL_SCHEMES` is patched, so we can use that.
```
>>> import distutils.command.install
>>> distutils.command.install.INSTALL_SCHEMES.keys()
dict_keys(['unix_prefix', 'unix_local', 'deb_system', 'unix_home', 'nt', 'nt_user', 'unix_user'])
```

The expansion there is less straight-forward, but we can make do. My only concern is distutils being deprecated and scheduled for removal in 3.12, we need to check how Debian/Ubuntu is planning to patch this.

Reply:
Ah, I remember now. The patch for sysconfig was never applied, *sigh*.

See https://github.com/pypa/pip/issues/9617.

We can either get the paths from distutils, or just use it as a way of flagging if the Python installation is patched by Debian.
I think the patch will now be applied in the next Debian version.

Reply:
@xclaesse sorry for the confusion, can you check the logic again and confirm it now works as expected on your system(s)?